docs/rules/prefer-istanbul-ignore-reason.md

```diff
@@ -0,0 +1,27 @@
+# Disallow disabling coverage for whole files (no-istanbul-ignore-file)
+
+This rule raises a warning about "istanbul ignore" comments missing a reason why
+code is ignored.
+
+## Rule Details
+
+The rule finds all comments matching the pattern `istanbul ignore <word>` and
+raises a warning if it is missing a reason afterwards.
+
+The following patterns are considered warnings:
+
+```js
+/* istanbul ignore file */
+
+// istanbul ignore file
```

Comment:
Should list ignore next/if/else here as well.

I would expect that this rule not block `istanbul ignore file: vendored from https://github.com/blah/blah` but `no-istanbul-ignore-file` would block that?